src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -182,6 +182,11 @@ def exchange(self, algorithm, peer_public_key):
                 _Reasons.UNSUPPORTED_EXCHANGE_ALGORITHM
             )
 
+        if type(peer_public_key.curve) is not type(self.curve):
```

Comment:
Yeah, I don't know what that means. What's teh right way to check if two curves rae equivilant.